### DISCUSSION TITLE
Storage GIT

### DISCUSSION THREAD

**Discussion by fdrubigny** [requarks/wiki#7191] [Error / Bug Report]
Bonjour tout le monde !
j'essaye d'activer le stockage Git sur mon installation de wiki js.
Je remplie les champs de configuration il me semble comme il faut.

Mais j'ai un bandeau rouge avec une erreur : "fatal: ni ceci ni aucun de ses répertoires parents n'est un dépôt git : .git" et je suis sûr de l'url du dépôt...
Que faire ?

Amicalement :)
-----
Hello everyone!
I'm trying to enable Git storage on my wiki js installation.
I fill in the configuration fields as it seems.

But I get a red banner with an error: "fatal: neither this nor any of its parent directories is a git repository: .git" and I'm sure of the repository url...
What can I do?

Best regards :)

<img width="694" alt="Capture d’écran 2024-05-26 à 10 48 22" src="https://github.com/requarks/wiki/assets/63860662/3acfb9cc-9b14-4eaf-b8c4-e2c54f3371cc">

**Comment by fnafox** [comment]
My guess is the local repo hasn't been created yet so there's nothing to push to remote git.
At the bottom of the git storage admin interface click `Purge Local Repository` then `Add Untracked Changes`, then try syncing again.

**Reply by fdrubigny** [reply]
Thank you for your reply.
I've tried but it doesn't make any difference, I always get the same error.
Best regards

**Reply by fnafox** [reply]
Does the wiki user (on the system) have permissions to create files at "Local Repository Path"?

**Reply by fdrubigny** [reply]
I think so, because the user is also the Wiki administrator. So I guess he has all the rights?
Best regards

**Reply by fnafox** [reply]
Oh I was asking on the host filesystem. Try using this path `./data/repo`

**Reply by fdrubigny** [reply]
Hello,
I'm already using this path, but I always get the same error...

**Reply by fnafox** [reply]
then I'm not sure without more info, my guess is that something is causing the repo to not get created locally which then later manifests as this problem

**Reply by Assan-zidan** [reply]
hello @fdrubigny 
Are you finaly slove your problem (enable GIT storage in wikijs) ? Because Ihave the same issue.

**Reply by fdrubigny** [reply]
Hello, sorry I don’t solve my problem

**Reply by stilobique** [reply]
Hello, you can access locally on your service ? Go here `/data/repo` and make a `git init`.

**Comment by harvthe2** [comment]
Here are the steps I used to connect to our local instance of gitlab. 

1. Log into the wikijs web interface as admin and click the Gear icon to get into the settings
2. Scroll down the left of the page to Storage
3. Check the box for Git
4. Ensure "This storage target" is a green "active"
5. Authentication type = ssh
6. Repository URL = git@yourgitlabinstance:yourrepository.git (or appropriate location)
7. Branch = Master (or appropriate branch and is Case sensitive)
8. SSH Private Key mode = path
9. A-SSH Private Key Path = /whereveryoustorekeys/id_rsa
10. B-SSH Private Key Contents = blank/empty
11. Verify SSL Cert = OFF (to the left)
12. Username = GitLab user ID
13. Password = GitLab password
14. Default Author Email = as appropriate for your instance
15. Default Author Name = wikijs
16. Local Repository Path = /var/wikijs/data/repo 
17. Always Locale Namespace = ON (to the right)
18. Git Binary Path = blank/empty
19. Sync Direction = Push to Target (this should push stuff from wikijs TO gitlab)
20. Every 0 Minute(s) 0 Hour(s) 1 Day(s) 0 Month(s) 0 Year(s) ---  as appropriate for your case
21. Click the green Apply button at the top of the page
22. Click the run option under "Add Untracked Changes"
23. Click the run option under "Force Sync"
24. Wait probably an hour to 90min for the first sync and hopefully at the bottom of the Storage Targets you will see a Status of Green Check for Git with the last time it was synchronized (we started with 1800ish pages and of course your machine/network speed will play into how long it takes)

Also I was getting odd errors when I had local storage and gitlab storage setup.  So try one or the other instead of both.
